prober/http.go

```diff
@@ -397,10 +400,23 @@ func ProbeHTTP(ctx context.Context, target string, module config.Module, registr
 	request = request.WithContext(ctx)
 
 	for key, value := range httpConfig.Headers {
-		if strings.Title(key) == "Host" {
+		normalizedKey := strings.Title(key)
+
+		if normalizedKey == "Host" {
 			request.Host = value
 			continue
 		}
+
+		// If there's a compression setting, and there's also an
+		// accept-encoding header, they MUST match, otherwise we
+		// end up requesting one encoding and trying to process
+		// a different one, which is more likely than not going
+		// to fail.
+		if httpConfig.Compression != "" && normalizedKey == "Accept-Encoding" && !strings.EqualFold(value, httpConfig.Compression) {
```

Comment:
Not correct: `Accept-Encoding: br;q=1.0, gzip;q=0.8, *;q=0.1`

Reply:
I was wondering about that case.

The intention of the change is to be very "do what I say". If I allow for "br;q=1.0, gzip;q=0.8" in principle the server can select whatever it wants from the acceptable choices. How would that case work? Or better said, what would the check be checking? For example, this configuration:

```
    compression: gzip
    headers:
        Accept-Encoding: br;q=1.0, gzip;q=0.8
```

if the server responds with br-encoded content, this configuration would still fail because it will try to decompress it as gzip. You can argue that the user is trying to verify that the server selects gzip even if the user says br is preferred.

I think that's an acceptable use case. I can remove the strict checking here, allow the user to specify any Accept-Encoding and still fail in the case described above. I should update the documentation to spell that out.

Would that be OK?

Reply:
I took a stab at this. I added code to process the Accept-Encoding header, so that the compression algorithm must be acceptable, so this configuration is valid:

```
    compression: gzip
    headers:
        Accept-Encoding: br;q=1.0, gzip;q=0.8
```

and this is invalid:

```
    compression: gzip
    headers:
        Accept-Encoding: br;q=1.0, deflate;q=0.8
```

the second configuration would cause the server to return something that will fail decompression.